docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,454 @@
+---
+title: "Migrate from Newtonsoft.Json to System.Text.Json - .NET"
+author: tdykstra
+ms.author: tdykstra
+ms.date: "12/20/2019"
+helpviewer_keywords: 
+  - "JSON serialization"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# How to migrate from Newtonsoft.Json to System.Text.Json
+
+This article shows how to migrate from [Newtonsoft.Json](https://www.newtonsoft.com/json) to [System.Text.Json](system-text-json-overview.md).
+
+ `System.Text.Json` focuses primarily on performance, security, and standards compliance. It has some key differences in default behavior and doesn't aim to have feature parity with `Newtonsoft.Json`. For some scenarios, `System.Text.Json` has no built-in functionality, but there are recommended workarounds. For other scenarios, workarounds are impractical. If your application depends on a missing feature, consider [filing an issue](https://github.com/dotnet/runtime/issues/new) to find out if support for your scenario can be added.
+
+<!-- For information about which features might be added in future releases, see the [Roadmap](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Text.Json/roadmap/README.md). [Restore this when the roadmap is updated.]-->
+
+Most of this article is about how to use the <xref:System.Text.Json.JsonSerializer> API, but it also includes guidance on how to use the <xref:System.Text.Json.JsonDocument> Document Object Model (DOM) and the <xref:System.Text.Json.Utf8JsonReader> and <xref:System.Text.Json.Utf8JsonWriter> API. The article is organized into sections in the following order:
+
+* [Differences in **default** JsonSerializer behavior compared to Newtonsoft.Json](#differences-in-default-behavior).
+* [Scenarios using JsonSerializer that require workarounds](#scenarios-using-jsonserializer-that-require-workarounds).
+* [Scenarios that JsonSerializer currently doesn't support](#scenarios-that-jsonserializer-currently-doesnt-support).
+* [JsonDocument](#jsondocument).
+* [Utf8JsonReader](#utf8jsonreader).
+* [Utf8JsonWriter](#utf8jsonwriter).
+
+## Differences in default behavior
```

Comment:
Make heading match the list above?

```suggestion
## Differences in default JsonSerializer behavior compared to Newtonsoft.Json
```

Reply:
I was thinking that would be unnecessarily long and what it refers to would be clear in context, especially immediately following the list which contains the extended description.  Also, in general the style guidance is to try to make h2 headings be short enough so they don't wrap. That said,  it's often impossible to keep h2's from wrapping and still keep them intelligible, and with the revised structure the number of h2's is limited, so I'll go with the longer form.